pkg/pillar/cmd/zedagent/attesttask.go

```diff
@@ -244,14 +246,31 @@ func (server *VerifierImpl) SendAttestQuote(ctx *zattest.Context) error {
 		return zattest.ErrControllerReqFailed
 	}
 
+	attestReq.Quote = quote
+
 	if attestCtx.EventLogParseErr == nil {
 		//On some platforms, either the kernel does not export TPM Eventlog
 		//or the TPM does not have SHA256 bank enabled for PCRs. We populate
 		//eventlog if we are able to parse eventlog successfully
-		encodeEventLog(attestCtx, quote)
-	}
+		encodeEventLog(attestCtx, attestReq.Quote)
 
-	attestReq.Quote = quote
+		if proto.Size(attestReq) > maxQuotePayloadSize {
+			log.Warnf("[ATTEST] attestReq size too much (%d), try to deduplicate", proto.Size(attestReq))
+			deduplicateEventLog(attestReq.Quote)
+		}
+
+		if proto.Size(attestReq) > maxQuotePayloadSize {
+			if len(attestReq.Quote.EventLog) > 0 {
+				log.Warnf("[ATTEST] attestReq size too much (%d), try to reduce event log (current count %d)", proto.Size(attestReq), len(attestReq.Quote.EventLog))
+				for proto.Size(attestReq) > maxQuotePayloadSize && len(attestReq.Quote.EventLog) > 0 {
+					attestReq.Quote.EventLog = attestReq.Quote.EventLog[1:]
+				}
```

Comment:
How do we know that this dropping from the front results in something which is useful when validating the event log?